README.md

```diff
@@ -2,58 +2,139 @@
+
+#### Usage Scenarios
+
+*  [In a Django web application](#django) The toolkit is fully integrated in Django framework. It contains utilities to mark your source strings for translation, **retrieve the translated strings** and display translation in templates or views. It also provides special CLI commands for pushing to TX and even migrating your existing template code to the new syntax.
+
+*  [In a Python application](#python) In other python applications or frameworks you can utilize the toolkit as a python library. The building blocks are there: you can push your source strings to TX and retrieve translations over the air to display them to your users.
+
+### Features
 
-# Generic Features - Supported Functionality
 This section refers to the generic functionality provided by the Toolkit. For implementation details refer to the specific usage scenario detailed in following sections.
 
 - **ICU message Format** (variables, plurals), context, metadata (comment, charlimit, tags) support
 - HTML escaping & un-escaping
 - Automatic fetch of translations over-the-air (OTA) via a background thread
 - Policies to handle missing translations or errors
-- Use of an intermediary service, [CDS](#hosting-translations-on-your-servers),to retrieve translations and an in-toolkit memory cache to serve them fast.
+- Use of an intermediary service, [CDS](#hosting-translations), to retrieve translations and an in-toolkit memory cache to serve them fast.
 
-# Installation
-Install the Toolkit to your project
-```python
-pip install transifex-python
-```
+### Authentication
 
-## Transifex Setup
 Before you begin using the Toolkit, you will also need an account in [Transifex](https://www.transifex.com) and a project.
+
 To set a project compatible with this toolkit contact [support](https://www.transifex.com/contact/) and you
 will be given a set of credentials (a public token and a secret), that you can use in your code for authentication.
+
 We will refer to these credentials in the text below as:
+
 - `project_token` (used for pulling translations from Transifex)
 - `project_secret` (used for pushing source content to Transifex)
 
+### Hosting Translations
+
+The Transifex Native solution retrieves the translated content via an intermediate called Content Delivery Service (CDS). It works similarly to a CDN and serves all translations from a cache, so that the retrieval is fast.
+
+We offer a cloud-based CDS instance at Transifex, however, you can (and we encourage you) to host it yourself, so that translations are served directly from your servers.
+In order to do that, you need to provide its host in the settings file of your project.
+
+For 
+
+```python
+TRANSIFEX_CDS_HOST = 'https://cds.example.com'
+```
+
+## Supported Frameworks
+
+### Django
 
-# Use-case Scenarios
+#### Installation
 
-## Scenario 1: Integrate with a Django app
+**Transifex Python Toolkit** supports all officially maintained versions of Python and Django.
```

Comment:
It doesn't yet, we want to though :D 
We support Python 2.7, 3.6 & Django 1.11

Reply:
Changed